### DISCUSSION TITLE
$request->has('photo') returns false when uploading a file.

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [laravel/framework#18735] [Q&A] [answered]
- Laravel Version: v5.3.31
- PHP Version: 5.6.24
- Database Driver & Version: N/A

### Description:
When I check to see if a request has an uploaded file, the method `request->has('photo');` returns `false` even when the request has a file.

I created a form with few inputs. One of the inputs, allows uploading file. In the method that handles the request, I did this following

`dd($request->all(), $request->has('photo'));`

The first output shows 'photo' but the second output returns false. Here is a screenshot.
![screenshot_1](https://cloud.githubusercontent.com/assets/24724371/24831756/0c056994-1c55-11e7-8c92-e31a51e5c9e2.png)

Not sure if this is a bug or intended to work this way. But this is what I have done instead

    $file = Input::file('photo');

    if( $file && $file->isValid() )
    {
        .....
    }

**Comment by caiquecastro** [comment]
Try $request->hasFile('photo')

**Reply by anturibit** [reply]
Im doing the same $request->hasFile('video') and sending it with header multipart/formdata using postman, if return video it looks like: "video": {} wich means indeed is receiving the data, but if i return return $request->hasFile('video') it returns false 
![image](https://github.com/user-attachments/assets/157365bb-2327-4c8e-a082-782f0b1f404f)

even if i use postman the result is the same 
![image](https://github.com/user-attachments/assets/79e3777d-663e-4b0b-9c04-e10d7eef922f)

**Comment by caiquecastro** [comment]
You should read the docs: https://laravel.com/docs/5.4/requests#files

**Reply by pradeep2011** [reply]
not working your code

**Comment by srmklive** [comment]
First read the docs. The correct method to do this is" `$request->hasFile('photo')`.

**Comment by haratmalli** [comment]
$request->hasFile('photo')...i m doing the same but same it returns false.

**Reply by haratmalli** [reply]
Make sure enctype="multipart/form-data" is there is your forn

**Reply by Burak-Ozcan-1** [reply]
thank you you are right 

enctype I misspelled a letter
form action="{{route('settings.Update',['id' => $settings->id])}}" method="post" enctype="multipart/form-data"

**Comment by lcardosozago** [comment]
I have the same problem... $request->hasFile('file') returning false

**Reply by haratmalli** [reply]
Make sure enctype="multipart/form-data" is there is your forn

**Comment by jonaswebdev** [comment, marked as answer]
Maybe can be the "PHP max upload file size", I had this here ;)

**Comment by nanyKuzola** [comment]
i'm getting the same error

**Reply by haratmalli** [reply]
Make sure enctype="multipart/form-data" is there is your forn

**Comment by joscript** [comment]
getting the same error $request->hasFile('file') returning false

**Reply by haratmalli** [reply]
Make sure enctype="multipart/form-data" is there is your forn

**Reply by sarvaiyaakshay** [reply]
> error $request->hasFile('file') returni

<form action="" method="post" enctype="multipart/form-data">

apply enctype="multipart/form-data" attribute in form tag

**Comment by rjruiz-dev** [comment]
Tengo el mismo error he cambiado el tamaño maximo a 2mb, incluso verificando dd($request-> hasFile ('user_photo'));
y devuelve false
alguna sugerencia?

**Comment by zerobity** [comment]
Igual, tengo configurado 20mb y no puedo cargar mas pequeñas.

**Comment by go2tec** [comment]
My request was **PUT** type, when I changed it to **POST** things worked as expected, can some one explain this ??!!

**Reply by haratmalli** [reply]
check route file
